File: firefly-example/src/main/java/com/firefly/example/reactive/coffee/store/router/impl/sys/StaticResourceHandler.java
package com.firefly.example.reactive.coffee.store.router.impl.sys;

import com.firefly.annotation.Component;
import com.firefly.codec.http2.model.HttpHeader;
import com.firefly.server.http2.router.Handler;
import com.firefly.server.http2.router.RoutingContext;
import com.firefly.server.http2.router.handler.file.StaticFileHandler;
import com.firefly.utils.exception.CommonRuntimeException;
import com.firefly.utils.log.slf4j.ext.LazyLogger;

import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.Arrays;
import java.util.List;

/**
 * @author Pengtao Qiu
 */
@Component("staticResourceHandler")
public class StaticResourceHandler implements Handler {

    public static final LazyLogger log = LazyLogger.create();

    private final List<String> staticResources = Arrays.asList("/favicon.ico", "/static/*");
    private final StaticFileHandler staticFileHandler;

    public StaticResourceHandler() {
        try {
            Path path = Paths.get(SysRouterInstaller.class.getResource("/").toURI());
            staticFileHandler = new StaticFileHandler(path.toAbsolutePath().toString());
        } catch (Exception e) {
            throw new CommonRuntimeException(e);
        }
    }

    @Override
    public void handle(RoutingContext ctx) {
        log.info(() -> "static file request -> " + ctx.getURI());
        ctx.put(HttpHeader.CACHE_CONTROL, "max-age=86400");
        staticFileHandler.handle(ctx);
    }

    public List<String> getStaticResources() {
        return staticResources;
    }

    public StaticFileHandler getStaticFileHandler() {
        return staticFileHandler;
    }
}

File: firefly-example/src/main/kotlin/com/firefly/example/kotlin/coffee/store/router/impl/sys/StaticResourceHandler.kt
package com.firefly.example.kotlin.coffee.store.router.impl.sys

import com.firefly.annotation.Component
import com.firefly.codec.http2.model.HttpHeader
import com.firefly.kotlin.ext.http.AsyncHandler
import com.firefly.kotlin.ext.http.AsyncStaticFileHandler
import com.firefly.kotlin.ext.log.KtLogger
import com.firefly.server.http2.router.RoutingContext
import com.firefly.utils.exception.CommonRuntimeException
import java.nio.file.Paths

/**
 * @author Pengtao Qiu
 */
@Component("staticResourceHandler")
class StaticResourceHandler : AsyncHandler {

    val staticResources = listOf("/favicon.ico", "/static/*")

    companion object {
        private val log = KtLogger.getLogger { }
        private val staticFileHandler = createStaticFileHandler()

        private fun createStaticFileHandler(): AsyncStaticFileHandler {
            try {
                val path = Paths.get(SysRouterInstaller::class.java.getResource("/").toURI())
                return AsyncStaticFileHandler(path.toAbsolutePath().toString())
            } catch (e: Exception) {
                throw CommonRuntimeException(e)
            }
        }
    }

    override suspend fun handle(ctx: RoutingContext) {
        log.info("static file request ${ctx.uri}")
        ctx.put(HttpHeader.CACHE_CONTROL, "max-age=86400")
        staticFileHandler.handle(ctx)
    }

}